### PR TITLE
Issue/504 add auth

### LINKED ISSUE #504: Auth0

### PR DESCRIPTION
**Pull request by braddf**
# Pull Request

## Description

Add Auth0 integration and corresponding endpoints/auth loading state/logout page:

image

Fixes #504

>N.B. also includes small fix for `nowcasting-app` as the SSR wasn't passing Cypress tests with date string formats being different between prerender on server and render on client-side, causing a React hydration error. Have forced the component to render client-side for now, but may need to keep an eye on this for date/time stuff.

## How Has This Been Tested?

Locally and Vercel preview branch: 
https://quartz-4su29j2w3-openclimatefix.vercel.app/

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by peterdudfield** [comment]
Node CI error?

File: apps/quartz-app/src/components/Main.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
+"use client";
```
Comment (peterdudfield) [inline apps/quartz-app/src/components/Main.tsx:1]: should this be `main.tsx` not `Main.tsx`

File: apps/quartz-app/app/api/auth/[auth0]/route.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { handleAuth, handleLogout } from "@auth0/nextjs-auth0";
+
```
Comment (peterdudfield) [inline apps/quartz-app/app/api/auth/[auth0]/route.ts:2]: could you just add comments of what this functions do?

File: apps/quartz-app/app/api/token/route.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { NextResponse } from "next/server";
+import { getAccessToken, withApiAuthRequired } from "@auth0/nextjs-auth0";
+
```
Comment (peterdudfield) [inline apps/quartz-app/app/api/token/route.ts:3]: doc string here would be great?

File: apps/quartz-app/src/components/Main.tsx (hunk 1 above)
Comment (braddf) [reply apps/quartz-app/src/components/Main.tsx:1]: Nah, React components are always capitalised by convention

File: apps/nowcasting-app/components/map/pvLatestMap.tsx [hunk 4]
```diff
@@ -17,8 +16,11 @@ import { FeatureCollection } from "geojson";
 import { safelyUpdateMapData } from "../helpers/mapUtils";
 import { components } from "../../types/quartz-api";
 import { generateGeoJsonForecastData } from "../helpers/data";
+import dynamic from "next/dynamic";
```
Comment (peterdudfield) [inline apps/nowcasting-app/components/map/pvLatestMap.tsx:19]: whats this doing?

**Review by peterdudfield — CHANGES_REQUESTED** [review]
Code looks great, and app seems to be working

Ive just left a few comments, do you mind looking at them?

**Comment by braddf** [comment]
> Node CI error?

@peterdudfield Yes – this needs a bit of tinkering to get Cypress to run properly in the CI – I think I got most of the way there but need to revisit when I get a chance... is an expected error atm and it runs pre-commit successfully so should be okay in meantime

File: apps/nowcasting-app/components/map/pvLatestMap.tsx (hunk 4 above)
Comment (braddf) [reply apps/nowcasting-app/components/map/pvLatestMap.tsx:19]: Yep this is the NB comment in PR description, it's a weird SSR thing I needed to fix to push the code

**Review by peterdudfield — APPROVED** [review]

**Pull request merged**